x-pack/plugins/security_solution/server/lib/risk_engine/tasks/helpers.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import datemath from '@kbn/datemath';
+import {
+  CoreKibanaRequest,
+  type KibanaRequest,
+  SECURITY_EXTENSION_ID,
+  type CoreStart,
+} from '@kbn/core/server';
+import { addSpaceIdToPath } from '@kbn/spaces-plugin/server';
+
+import type { Range } from '../../../../common/risk_engine';
+
+export const convertDateToISOString = (dateString: string): string => {
+  const date = datemath.parse(dateString);
+
+  if (date?.isValid()) {
+    return date.toISOString();
+  } else {
+    throw new Error(`Could not convert string "${dateString}" to ISO string`);
+  }
+};
+
+export const convertRangeToISO = (range: Range): Range => ({
+  start: convertDateToISOString(range.start),
+  end: convertDateToISOString(range.end),
+});
+
+const buildFakeScopedRequest = ({
+  coreStart,
+  namespace,
+}: {
+  coreStart: CoreStart;
+  namespace: string;
+}): KibanaRequest => {
+  const rawRequest = {
+    headers: {},
+    path: '/',
+  };
+
+  const request = CoreKibanaRequest.from(rawRequest);
+  const scopedPath = addSpaceIdToPath('/', namespace);
+
+  coreStart.http.basePath.set(request, scopedPath);
+
+  return request;
+};
+
+export const buildScopedInternalSavedObjectsClient = ({
```

Comment:
__nit:__ Could you please consider adding a comment for future readers to explain the purpose of this method, the reason for disabling the security extension, and why this method should be used sparingly (e.g., only in scenarios where there's no access to the user context, such as within a background task)? While it may seem obvious now, it's easy to forget these details after a few months, and the comment can serve as a helpful reminder. Additionally, it's our practice to prefix such methods with `Unsafe` or `Insecure` to clearly indicate that they should be used with caution.
```suggestion
export const buildScopedInternalSavedObjectsClientUnsafe = ({
```

Reply:
Will do! I still have the habit of putting that context in commit messages, but our squashing strategy makes that much less useful 😅 .

Reply:
Addressed in https://github.com/elastic/kibana/pull/163216/commits/80785e2ee3ec48b99e5d70be5956625dacd11d95.